arangod/RestHandler/RestDumpHandler.cpp

```
@@ -0,0 +1,269 @@
+      return ResultT<std::pair<std::string, bool>>::error(
+          TRI_ERROR_BAD_PARAMETER,
+          "expecting POST /_api/dump/start or /_api/dump/next/<id>");
+    }
+  } else {
+    // invalid HTTP method
+    return ResultT<std::pair<std::string, bool>>::error(
+        TRI_ERROR_HTTP_METHOD_NOT_ALLOWED);
+  }
+
+  if (ServerState::instance()->isCoordinator()) {
+    ServerID const& DBserver = _request->value("dbserver");
+    if (!DBserver.empty()) {
+      // if DBserver property present, remove auth header
+      return std::make_pair(DBserver, true);
+    }
+
+    return ResultT<std::pair<std::string, bool>>::error(
+        TRI_ERROR_BAD_PARAMETER, "need a 'dbserver' parameter");
+  }
+
+  return {std::make_pair(StaticStrings::Empty, false)};
+}
+
+void RestDumpHandler::handleCommandDumpStart() {
+  std::string database = _request->databaseName();
+  std::string user = _request->user();
+
+  // TODO: check permissions
+
+  bool parseSuccess = false;
+  VPackSlice body = this->parseVPackBody(parseSuccess);
+  if (!parseSuccess) {  // error message generated in parseVPackBody
+    return;
+  }
+
+  if (!body.isObject()) {
+    generateError(Result(TRI_ERROR_BAD_PARAMETER, "invalid request body"));
+    return;
+  }
+
+  RocksDBDumpContextOptions opts;
+  opts.batchSize = [&body]() {
+    if (auto s = body.get("batchSize"); s.isNumber()) {
+      return s.getNumber<uint64_t>();
+    }
+    return ::defaultBatchSize;
+  }();
+  opts.prefetchCount = [&body]() {
+    if (auto s = body.get("prefetchCount"); s.isNumber()) {
+      return s.getNumber<uint64_t>();
+    }
+    return ::defaultPrefetchCount;
+  }();
+  opts.parallelism = [&body]() {
+    if (auto s = body.get("parallelism"); s.isNumber()) {
+      return s.getNumber<uint64_t>();
+    }
+    return ::defaultParallelism;
+  }();
+  opts.ttl = [&body]() {
+    if (auto s = body.get("ttl"); s.isNumber()) {
+      return s.getNumber<double>();
+    }
+    return ::defaultTtl;
+  }();
```

What if any of those parameters is present but not a number? It looks like we will then silently ignore it. Shouldn't we reject the request with an error instead?